sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HistogramNumeric.scala

```diff
@@ -72,6 +74,8 @@ case class HistogramNumeric(
     case n: Int => n
   }
 
+  private def propagateInputType: Boolean = SQLConf.get.histogramNumericPropagateInputType
```

Comment:
This is another tricky thing: the semantic/behavior of an expression should be static, not dynamic that can be changed in the middle at runtime. Think about a user who creates a view with this function, then he/she read the view back, the behavior should be the same with when the view was created, no matter what the config is currently.

That's why we have `case class Size(child: Expression, legacySizeOfNull: Boolean)`: we put the flag in constructor, so that the semantic/behavior is fixed once the expression is created.

Reply:
Thanks, that's a good point. Updated to make this static.